src/eos-metrics-instrumentation.c

```diff
@@ -284,6 +300,70 @@ record_live_boot (gpointer unused)
     return G_SOURCE_REMOVE;
 }
 
+static gchar *
+get_image_version_for_path (const gchar *path)
+{
+    ssize_t xattr_size = 0;
```

Comment:
Nitpick: Misleading to assign a value to this at the moment. If you feel you should, I suggest it’s `-1` to indicate an invalid xattr, rather than `0` to indicate a present-but-zero-length one.